Framework/NexusGeometry/inc/MantidNexusGeometry/JSONInstrumentBuilder.h

```diff
@@ -0,0 +1,40 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_NEXUSGEOMETRY_JSONINSTRUMENTBUILDER_H_
+#define MANTID_NEXUSGEOMETRY_JSONINSTRUMENTBUILDER_H_
+
+#include "MantidGeometry/Instrument_fwd.h"
+#include "MantidNexusGeometry/DllConfig.h"
+#include "MantidNexusGeometry/JSONGeometryParser.h"
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace Mantid {
+namespace NexusGeometry {
+/** JSONInstrumentBuilder : Builds in-memory instrument from json string
+ * representing Nexus instrument geometry.
+ */
+class MANTID_NEXUSGEOMETRY_DLL JSONInstrumentBuilder {
+public:
+  JSONInstrumentBuilder() = delete;
+  JSONInstrumentBuilder(const std::string &jsonGeometry);
```

Comment:
It's a good idea to mark single arg constructors as `explicit` to avoid unintended implicit conversions.